.travis.yml

```diff
@@ -25,87 +23,51 @@ addons:
   homebrew:
     packages:
       - git
-  apt:
-    sources:
-      - llvm-toolchain-xenial-8
-      - ubuntu-toolchain-r-test
-    packages:
-      - llvm-8
-      - g++-9
-      - clang-8
-      - git
-      - python3
-      - python3-pip
-      - python3-setuptools
-      - clang-tidy-8
+
+services:
+  - docker
 
 cache:
-  apt: true
   cargo: true
   ccache: true
   pip: true
   directories:
     - "$HOME/.ccache"
     - "$HOME/Library/Caches/Homebrew"
 
-matrix:
+jobs:
   include:
     - os: osx
       name: "OSX AppleClang"
       if: type = pull_request OR branch = master
       osx_image: xcode11
+      install: ./housekeeping/macos/dependency.sh
       before_cache:
         - brew cleanup
+      env:
+        - BUILD_DIR=build
+        - BUILD_TREADS=2
+      script:
+        - ./housekeeping/makeBuild.sh  -DCOVERAGE=${COVERAGE:-OFF} -DCMAKE_TOOLCHAIN_FILE=${TOOLCHAIN:-cmake/toolchain/cxx17.cmake}
 
     - os: linux
       name: "Linux gcc-9 clang-tidy"
       if: type = pull_request OR branch = master
       env:
-        - _CC=gcc-9
-        - _CXX=g++-9
+        - BUILD_TARGET=generated
+        - BUILD_DIR=build
+        - BUILD_TREADS=2
+        - INDOCKER_IMAGE=soramitsu/kagome-dev:8
       script:
-        - export CC=${_CC}
-        - export CXX=${_CXX}
-        - export PATH="$HOME/.cargo/bin:$PATH"
-        - export LLVM_ROOT=/usr/lib/llvm-8
-        - export PATH=${LLVM_ROOT}/bin:${LLVM_ROOT}/share/clang:${PATH}
-        - sudo update-alternatives --install /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-8 90
-        - cmake . -Bbuild
-        # build only generated files, so clang-tidy will work correctly
-        - cmake --build build --target generated
-        - housekeeping/clang-tidy.sh build
+        - ./housekeeping/indocker.sh ./housekeeping/makeBuild.sh
+        - ./housekeeping/indocker.sh ./housekeeping/clang-tidy.sh "${BUILD_DIR}"
 
-install:
-  - |
-    # install python pip3 deps
-    if [ "$TRAVIS_OS_NAME" = "linux" ]; then
-        LLVM_BIN=/usr/lib/llvm-8/bin
-        if [ -d $LLVM_BIN ]; then
-          export PATH="$PATH:$LLVM_BIN"
-          echo "$LLVM_BIN added to PATH"
-        else
-          echo "$LLVM_BIN is absent"
-        fi
-      elif [ "$TRAVIS_OS_NAME" = "osx" ]; then
-        pip3 install --user pyyaml
-      fi
-    sudo python3 -m pip install --upgrade pip
-    sudo python3 -m pip install scikit-build
-    sudo python3 -m pip install cmake requests gitpython gcovr
-  - export PATH="$HOME/.cargo/bin:$PATH"
-  - curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain nightly-2019-10-16 --profile minimal
 
 before_script:
-  - cmake --version
-  - python -V
-  - python3 -V
-  - pip3 -V
+  - cmake --version || true
+  - python -V || true
+  - python3 -V || true
+  - pip3 -V || true
   - gcovr --version || true
-  - echo $PATH
+  - echo $PATH || true
```

Comment:
For what it if this will be doing in docker container?

Reply:
I think it was used mostly  for mac, I did not clean it because we will remove .travis.yml in next PR.